### PR TITLE
various fixes to transaction list

### LINKED ISSUE #8686: Tx History View More button renders with fewer than 10 txs.
Seems that the View More Button with render the first 10 txs, but there's no minimal limit that renders with. Such as if only an account's history has fewer than 10 txs it will still render the View More button.

<details>
<summary>Issue with View More Button with only 1 transaction</summary>

</details>

### PR DESCRIPTION
**Pull request by brad-decker**
1 - Hides the activity log in the transaction details if there are no activities
2 - Fixes various PropTypes issues caused by accidental import of the component instead of the container for Transaction details
3 - Fixes paging logic for history

@Gudahtt i am still getting one PropType issue about recipient address on contract deployments, and I'm trying to track down the logic for how it was handled before. IF I look at the initialTransaction for any contractDeployment transaction group it doesn't have a 'to' in the txParams. If i view the transaction on etherscan, it has the newly deployed contract address and I think that matches to the contractAddress field in txReciept but I don't see that logic being previously employed in Transaction Details. Do you have any insights into this?

Fixes #8686

### CODE REVIEW

File: ui/app/components/app/transaction-list-item/transaction-list-item.component.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ export default function TransactionListItem ({ transactionGroup, isEarliestNonce
   } else if (isReceive) {
     Icon = Receive
   } else if (isInteraction) {
+    console.log(transactionGroup)
```
Comment (whymarrh) [inline ui/app/components/app/transaction-list-item/transaction-list-item.component.js:77]: Nit

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: ui/app/components/app/transaction-list-item/transaction-list-item.component.js (hunk 1 above)
Comment (brad-decker) [reply ui/app/components/app/transaction-list-item/transaction-list-item.component.js]: Oops. Removed the console

**Review by whymarrh — DISMISSED** [review]
LGTM!

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
>
@Gudahtt i am still getting one PropType issue about recipient address on contract deployments, and I'm trying to track down the logic for how it was handled before. IF I look at the initialTransaction for any contractDeployment transaction group it doesn't have a 'to' in the txParams. If i view the transaction on etherscan, it has the newly deployed contract address and I think that matches to the contractAddress field in txReciept but I don't see that logic being previously employed in Transaction Details. Do you have any insights into this?

I suspect the old transaction list didn't show a recipient at all. We can't know the resulting contract address until after the transaction has completed (afaik).

**Comment by brad-decker** [comment]
Removed that requirement on the proptype so that should resolve all the warnings.

**Comment by whymarrh** [comment]
> Removed that requirement on the proptype so that should resolve all the warnings.

We'll need to update all the places the component used `recipientAddress` to make sure they're not expecting a value.

**Comment by Gudahtt** [comment]
^ I had thought it was only used in one place - the `SenderToRecipient` component, which already considered `recpientAddress` to be optional.

It looks like this would also blow up on the `tryReverseResolveAddress` call though 🤔 So that should be skipped if there's no `recipientAddress`

**Comment by Gudahtt** [comment]
Oh, also in the `transaction-list-item-details` container. I missed that.

**Comment by brad-decker** [comment]
> We'll need to update all the places the component used recipientAddress to make sure they're not expecting a value.

It is used in two places, the container and passed to the SenderToRecipient component, which doesn't require a recipientAddress. 

I'll update the logic in the container to not attempt to do anything with the recipientAddress if it is undefined.

**Comment by brad-decker** [comment]
I mean, it doesn't blow it up. 😬 It's just not necessary

**Comment by brad-decker** [comment]
There's going to be more prop types changes as a result of this work. recipientNickname is only defined if recipientAddress is defined. etc

**Comment by brad-decker** [comment]
image
correction it doesn't blow up thanks to error handling :P

File: ui/app/css/itcss/components/newui-sections.scss [hunk 2]
```diff
@@ -78,7 +78,6 @@ $sub-mid-size-breakpoint-range: "screen and (min-width: #{$break-large}) and (ma
 
   .main-container {
     width: 100%;
-    overflow-y: auto;
```
Comment (Gudahtt) [inline ui/app/css/itcss/components/newui-sections.scss:81]: What prompted this change?

Comment (brad-decker) [reply ui/app/css/itcss/components/newui-sections.scss:81]: it shouldn't be there, I was switching between branches and trying to debug the issue with disappearing transaction history on firefox. Sorry @Gudahtt

Comment (brad-decker) [reply ui/app/css/itcss/components/newui-sections.scss]: removed

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**